.github/workflows/external-application-connection-tests.yml

```diff
@@ -0,0 +1,203 @@
+
+      - name: Create network
+        run: docker network create example
+
+      - name: Run container
+        run: |
+          tests/bin/runner-init.sh
+        env:
+          IMAGE: jss-runner
+          NAME: pki
+          HOSTNAME: pki.example.com
+            
+      - name: Connect PKI container to network
+        run: docker network connect example pki --alias pki.example.com
+
+      - name: Install DS and PKI packages
+        run: docker exec pki dnf install -y 389-ds-base pki-ca pki-tests dogtag-pki-acme
+
+      - name: Install DS
+        run: docker exec pki ${SHARED}/tests/bin/ds-create.sh
+
+      - name: Install CA
+        run: |
+          docker exec pki pkispawn \
+              -f /usr/share/pki/server/examples/installation/ca.cfg \
+              -s CA \
+              -D pki_cert_id_generator=random \
+              -D pki_request_id_generator=random \
+              -v
+
+      - name: Run PKI healthcheck
+        run: docker exec pki pki-healthcheck --debug
+
+      - name: Initialize PKI client
+        run: |
+          docker exec pki pki-server cert-export ca_signing --cert-file ca_signing.crt
+          docker exec pki pki client-cert-import ca_signing --ca-cert ca_signing.crt
+          docker exec pki pki info
+
+      - name: Check CA admin
+        run: |
+          docker exec pki pki client-cert-import \
+              --pkcs12 /root/.dogtag/pki-tomcat/ca_admin_cert.p12 \
+              --pkcs12-password Secret.123
+          docker exec pki pki -n caadmin ca-user-show caadmin
+
+      - name: Create postgresql certificates
+        run: |
+          docker exec pki pki nss-cert-request \
+              --subject "CN=postgresql.example.com" \
+              --ext /usr/share/pki/server/certs/sslserver.conf \
+              --csr sslserver.csr
+          REC_ID=$(docker exec pki pki ca-cert-request-submit  \
+              --profile caServerCert \
+              --csr-file sslserver.csr \
+              --subject "CN=postgresql.example.com" | grep "Request ID")
+          CERT_ID=$(docker exec pki pki -n caadmin ca-cert-request-approve ${REC_ID:14} --force | \
+              grep "Certificate ID")
+          docker exec pki pki ca-cert-export ${CERT_ID:18} --output-file sslserver.crt
+          docker exec pki pki nss-cert-import --cert sslserver.crt postgresql
+          docker exec pki pk12util -o sslserver.p12 -n postgresql -d /root/.dogtag/nssdb -W secret
+          docker cp pki:sslserver.p12 .
+          openssl pkcs12 -in sslserver.p12 -nocerts -out sslserver.key -noenc -password  pass:secret
+          openssl pkcs12 -in sslserver.p12 -nokeys -clcerts -out sslserver.crt  -password pass:secret
+          openssl pkcs12 -in sslserver.p12 -nokeys -cacerts -out ca.crt  -password pass:secret
+
+      - name: Build postgrsql image with certificates
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          tags: postgres-ssl
+          target: postgres-ssl
```

Comment:
It looks like it's possible to deploy the standard PostgreSQL container with SSL (without building a new image):
* https://gist.github.com/mrw34/c97bb03ea1054afb551886ffc8b63c3b
* https://stackoverflow.com/questions/55072221/deploying-postgresql-docker-with-ssl-certificate-and-key-with-volumes

But I have not tried that myself so I'll let you decide.

Reply:
I have already tested this but I could not get it working without root access on my machine. If I try as normal user the files inside the container does not get the right ownership and postgrsql do not start.

Reply:
That's OK, we probably can revisit this another time.